documentation/private-link-service/private-link-redirect-map.md

```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * @brief Range matched key dst_port_range
+     */
+    sai_u32_range_t dst_port_range;
+
+    /**
+     * @brief Rule priority in table
+     */
+    sai_uint32_t priority;
+
+} sai_outbound_port_map_port_range_entry_t;
+```
+
+ It includes the following attributes:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_ACTION | sai_outbound_port_map_port_range_entry_action_t | Action. |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_IP | sai_ip_address_t | Back end IP. |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_MATCH_PORT_BASE | sai_uint16_t | Match port base. |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_PORT_BASE | sai_uint16_t | Back end port base. |
+| SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_COUNTER_ID | sai_object_id_t | Attach a counter. |
+
+The entry attribute **Action** is defined as below:
+```c
+typedef enum _sai_outbound_port_map_port_range_entry_action_t
+{
+    SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ACTION_SKIP_MAPPING,
+    SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ACTION_MAP_TO_PRIVATE_LINK_SERVICE,
+} sai_outbound_port_map_port_range_entry_action_t;
+```
+
+### 5.3. DASH CA-PA mapping attributes
+
+The following attributes will be added to CA-to-PA entry, for supporting service port rewrites for PL/PL NSG:
+
+| Attribute name | Type | Description |
+| --- | --- | --- |
+| SAI_OUTBOUND_CA_TO_PA_ENTRY_ATTR_OUTBOUND_PORT_MAP_ID | sai_object_id_t | Outbound port map id. |
+
+## 6. DASH pipeline behavior
+Following [DASH pipeline behavior of private link service](https://github.com/sonic-net/DASH/blob/main/documentation/private-link-service/private-link-service.md#6-dash-pipeline-behavior), it adds updates for redirect map.
+
+### 6.1  VM-to-PLS direction (Outbound)
+
+**Mapping - VNET**:
+Each mapping will be associated with a port map object:
+   | SAI field name | Type | Value |
+   | --- | --- | --- |
+   | entry.port_map_id | `sai_object_id_t` | (SAI object ID of the port map) |
+
+Later in the stage of outbound port map, the port map id and inner packet destination port consists of a match key to look up table outbound_port_map_port_range, the matched table entry contains a rewrite info for the redirect map.
+   | SAI field name | Type | Value |
+   | --- | --- | --- |
+   | entry.outbound_port_map_id | `sai_object_id_t` | (SAI object ID of the port map) |
+   | entry.dst_port_range | `sai_u32_range_t` | 2000-2100 |
+   | entry_attr.SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_ACTION | `sai_outbound_port_map_port_range_entry_action_t` | `SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ACTION_MAP_TO_PRIVATE_LINK_SERVICE` |
+   | entry_attr.SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_IP | `sai_ip_address_t` | `3.3.3.1` |
+   | entry_attr.SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_MATCH_PORT_BASE | `sai_uint16_t` | `2000` |
+   | entry_attr.SAI_OUTBOUND_PORT_MAP_PORT_RANGE_ENTRY_ATTR_BACKEND_PORT_BASE | `sai_uint16_t` | `11000` |
+
```

Comment:
better add a clarification of metering. it follows the CA-PA mapping one, and currently doesn't have any special action in port map.

Reply:
Updated